### PR TITLE
feat: Add comprehensive tests for Ratio struct

### PR DESCRIPTION
**Pull request by Dayuxiaoshui**
Summary of Added Tests
A comprehensive integration test suite was added for the Ratio struct. This suite validates the correctness and robustness of the implementation by covering the following key areas:

Equivalence and Simplification: Tests to confirm that different but mathematically equivalent ratios (e.g., 1/2 and 2/4) are treated as equal, verifying the PartialEq implementation.

Inverse Operations: Validation of the inverse() method, including the correct handling of standard ratios, inverse-of-an-inverse chains, and the special case for Ratio::zero().

Multiplication Correctness: Detailed comparison between mul_trunc (truncation) and mul_round (rounding to the nearest integer) using various inputs to verify arithmetic accuracy.

Zero Value Behavior: Ensuring consistent and correct behavior for all types of zero ratios, including those created via Ratio::zero() and Ratio::new(0, x).

Panic Conditions: Verification that the constructor and other methods panic on invalid input (e.g., division by zero for a non-zero numerator), as specified in the documentation. This is achieved using the #[should_panic] attribute.

Edge Case Analysis: Testing with u32::MAX to check for potential overflows and confirm high precision, especially for ratios equal to or very close to 1.

Effect of the Tests
All tests pass successfully. This demonstrates that the Ratio implementation is robust, numerically stable, and handles edge cases correctly. The high precision of the underlying fixed-point calculation was also confirmed. This test suite provides high confidence in the code's correctness and helps prevent future regressions by ensuring that any subsequent changes do not break existing functionality.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -73,8 +92,11 @@ impl Ratio {
     ///
     /// let ratio = Ratio::new(1, 2);
     /// assert_eq!(ratio.inverse(), Ratio::new(2, 1));
+    ///
+    /// let zero = Ratio::zero();
+    /// assert_eq!(zero.inverse(), Ratio::zero());
```
Comment (equation314) [inline src/lib.rs:103]: This example is confusing, remove it or add comments.

File: src/lib.rs [hunk 2]
```diff
@@ -19,6 +20,20 @@ pub struct Ratio {
...
+    /// the inverse ratio. Instead, it returns another zero ratio.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use int_ratio::Ratio;
+    ///
+    /// let zero = Ratio::zero();
+    /// assert_eq!(zero.mul_trunc(123), 0);
+    /// assert_eq!(zero.inverse(), Ratio::zero());
```
Comment (equation314) [inline src/lib.rs:38]: This example is confusing, remove it or add comments.

File: src/lib.rs [hunk 3]
```diff
@@ -73,8 +92,11 @@ impl Ratio {
     ///
     /// let ratio = Ratio::new(1, 2);
     /// assert_eq!(ratio.inverse(), Ratio::new(2, 1));
+    ///
+    /// let zero = Ratio::zero();
+    /// assert_eq!(zero.inverse(), Ratio::zero());
     /// ```
-    pub const fn inverse(&self) -> Self {
+    pub const fn inverse(self) -> Self {
```
Comment (equation314) [inline src/lib.rs]: Why change `&self` to `self`? Is there any performance improvement?

Comment (Dayuxiaoshui) [reply src/lib.rs]: Because passing by value allows the compiler to place the small struct directly into high-speed CPU registers, avoiding the overhead of memory access via a pointer.

File: src/lib.rs (hunk 1 above)
Comment (Dayuxiaoshui) [reply src/lib.rs:103]: Ok, working on adding comments

File: src/lib.rs (hunk 2 above)
Comment (Dayuxiaoshui) [reply src/lib.rs:38]: Ok, working on adding comments

File: src/lib.rs (hunk 3 above)
Comment (equation314) [reply src/lib.rs]: Ok, I forgot these changes had been merged.

**Comment by equation314** [comment]
Please fix cargo clippy warnings.

**Comment by Dayuxiaoshui** [comment]
> Please fix cargo clippy warnings.

ok,Resolved

File: src/lib.rs [hunk 4]
```diff
@@ -5,11 +5,11 @@ use core::{cmp::PartialEq, fmt};
 
 /// The ratio type.
 ///
-/// It converts `numerator / denominator` to `mult / (1 << shift)` to avoid
-/// `u128` division on calculation. The `shift` is  as large as possible to
+/// It converts numerator / denominator to mult / (1 << shift) to avoid
```
Comment (equation314) [inline src/lib.rs]: Why remove ``? It can make the generated rustdoc more readable.

File: src/lib.rs [hunk 5]
```diff
@@ -90,9 +93,12 @@ impl Ratio {
     /// ```
     /// use int_ratio::Ratio;
     ///
+    /// // The inverse of a standard ratio.
     /// let ratio = Ratio::new(1, 2);
     /// assert_eq!(ratio.inverse(), Ratio::new(2, 1));
     ///
+    /// // `Ratio::zero()` is a special case representing 0/0. Its inverse is defined
```
Comment (equation314) [inline src/lib.rs]: The use of `` is inconsistent.

Comment (Dayuxiaoshui) [reply src/lib.rs]: Resolved

Comment (equation314) [reply src/lib.rs]: I mean `Ratio::zero()` and `0/x` is contained by the `` quotes, but `0/0` is not.

Comment (Dayuxiaoshui) [reply src/lib.rs]: In the new commit, I've added 0/0 on ''
